woocommerce/woocommerce-ios
# [Order Details] Reuse payment methods from Order Details

Closes: #6782 
Part of: #7003 



### Description


This change renames and reuses the Simple Payments Payment Method screen from Order Details.

Merchants will now see a `Collect Payment` button for any unpaid order even if they aren't set up for Card-Present Payments. Currently, they still need to be Card-present eligible to see the `Collect Payment` button, but that requirement will be removed in a future PR.

Tapping Collect Payment shows the Payment Method screen and allows them to choose between Cash, Payment Link, and (if available) Card payment.

### Testing instructions


Using a US store with WCPay or Stripe

1. Go to the Orders tab, and add a new order using the + button
2. Use `Create Order` and set up the order
3. At the end of the flow, you'll be on the Order Details screen and should see `Collect Payment`
4. Tap `Collect Payment` and observe that you are shown the payment selection screen
5. Tap `Card` and take a card payment.
6. Observe that you're returned to the Order Details screen with the order paid when this is complete

N.B there is an outstanding issue where the `Collect Payment` button remains visible after the payment flow is complete. This will be addressed in the next PR.

Repeat the above using `Card` and `Payment Link`. Note that `Payment Link` does not mark the order as paid, as it requires further action by the customer to follow the link and make payment.

### Screenshots



https://user-images.githubusercontent.com/2472348/174629140-7108171b-1158-4a34-949a-d5e8aecb61ef.mp4


---
